PHPCompatibility/Tests/FunctionNameRestrictions/RemovedMagicMethodsUnitTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2020 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Tests\FunctionNameRestrictions;
+
+use PHPCompatibility\Tests\BaseSniffTestCase;
+
+/**
+ * Test the RemovedMagicMethodsSniff sniff.
+ *
+ * @group removedMagicMethods
+ * @group functionNameRestrictions
+ *
+ * @covers \PHPCompatibility\Sniffs\FunctionNameRestrictions\RemovedMagicMethodsSniff
+ *
+ * @since 10.0.0
+ */
+final class RemovedMagicMethodsUnitTest extends BaseSniffTestCase
+{
+
+    /**
+     * Ensure a warning message when found in versions that sof-deprecated the magic method.
+     *
+     * @dataProvider dataSoftDeprecations
+     *
+     * @param string $method      Method name
+     * @param string $alternative Alternative proposed
+     * @param int[]  $lineNumbers The line numbers of the violation
+     *
+     * @return void
+     */
+    public function testSoftDeprecations($method, $alternative, $lineNumbers)
+    {
+        $file            = $this->sniffFile(__FILE__, '8.5');
+        $expectedMessage = sprintf(
+            'Magic method %s is maintained for backward compatibility since PHP 8.5. Use %s instead.',
+            $method,
+            $alternative
+        );
+        foreach ($lineNumbers as $lineNumber) {
+            $this->assertWarning($file, $lineNumber, $expectedMessage);
+        }
+    }
+
+    /**
+     * Data provider.
+     *
+     * @see testSoftDeprecations()
+     *
+     * @return array
+     */
+    public static function dataSoftDeprecations()
+    {
+        return [
+            ['__sleep()', '__serialize', [47, 53, 59, 64]],
+            ['__wakeup()', '__unserialize', [48, 54, 60, 65]],
+        ];
+    }
+
+    /**
+     * Verify there are no false positives on valid code.
+     *
+     * @dataProvider dataNoFalsePositives
+     *
+     * @param string $method      Method name
+     * @param int[]  $lineNumbers The line numbers of the violation
+     *
+     * @return void
+     */
+    public function testNoFalsePositives($method, $lineNumbers)
```

Comment:
The `$method` parameter is unused and doesn't seem to have any value (other than documentation). Why does this exist ?